contingency/contingency-api/src/main/java/com/powsybl/contingency/contingency/list/identifier/IdBasedNetworkElementIdentifier.java

```diff
@@ -6,34 +6,45 @@
  */
 package com.powsybl.contingency.contingency.list.identifier;
 
+import com.google.common.collect.ImmutableSet;
 import com.powsybl.iidm.network.Identifiable;
 import com.powsybl.iidm.network.Network;
 
-import java.util.Objects;
-import java.util.Optional;
+import java.util.*;
+import java.util.stream.Collectors;
 
 /**
  * @author Etienne Lesot <etienne.lesot@rte-france.com>
  */
 public class IdBasedNetworkElementIdentifier implements NetworkElementIdentifier {
 
-    private final String identifier;
+    private final Set<String> identifiers;
```

Comment:
To me, it looks like you fixed the problem of supporting N-K defaults in NetworkElementIdentifierList class with : 
 `public Set<Identifiable<?>> filterIdentifiable(Network network) {
        Set<Identifiable<?>> identifiables = new LinkedHashSet<>();
        networkElementIdentifiers.forEach(identifiant -> identifiables.addAll(identifiant.filterIdentifiable(network)));
        return identifiables;
    }`

And here, we can keep a String identifier

Reply:
fixed